### ISSUE TITLE
Bitcode is not included in built .ipa file, even with include_bitcode: true

### ISSUE THREAD

**Issue by EskiMag** [fastlane/fastlane#8750]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Hi all, I am using `gym` for building and exporting the IPA of our app. Our app consists of main app, widget, photo extension and  watchapp. Everything works correctly, but when I am uploading the IPA to the iTunes Connect through deliver or application loader, I am getting an email from iTunes Connect containing this: **Invalid Executable - The executable '********.app/Watch/WatchKitApp.app/PlugIns/WatchKitApp Extension.appex/WatchKitApp Extension' does not contain bitcode.**

I have set the `ENABLE_BITCODE = YES;` build settings for watchapp and also watchkitextension. I am using `include_bitcode: true` option for `gym` command. Do you have any idea, what can be wrong? I also tried adding `-fembed-bitcode` into the `OTHER_CFLAGS` part of build configuration of watchapp and also watchkit extension.
Weird thing is, that regarding the watchapp Apple has no issues, but the watchkitextension is the problem for them.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

### Captured Output

Command Used: `appstore region:ausnz branch:********`
<details><summary>Output/Log</summary>

```

DEBUG [2017-04-03 12:54:52.24]: Checking if there are any plugins that should be loaded...
DEBUG [2017-04-03 12:54:52.26]: Using region: ********
DEBUG [2017-04-03 12:54:52.26]: Using branch: ********
ERROR [2017-04-03 12:54:52.27]: Name of the lane 'appstore' is already taken by the action named 'appstore'
INFO [2017-04-03 12:54:52.27]: Driving the lane 'appstore' 🚀
INFO [2017-04-03 12:54:53.82]: ----------------------------------
INFO [2017-04-03 12:54:53.82]: --- Step: Switch to build lane ---
INFO [2017-04-03 12:54:53.82]: ----------------------------------
INFO [2017-04-03 12:54:53.83]: Cruising over to lane 'build' 🚖
INFO [2017-04-03 12:54:58.69]: -----------------
INFO [2017-04-03 12:54:58.70]: --- Step: gym ---
INFO [2017-04-03 12:54:58.70]: -----------------
INFO [2017-04-03 12:54:58.84]: $ xcodebuild -list -project Main/Sources/Application/Projects/iOS/********.xcodeproj -configuration Default
INFO [2017-04-03 12:55:01.10]: $ xcodebuild clean -showBuildSettings -scheme ******** -project Main/Sources/Application/Projects/iOS/********.xcodeproj -configuration Default

+----------------------+-------------------------------------------------------------------------+
|                                     Summary for gym 2.24.0                                     |
+----------------------+-------------------------------------------------------------------------+
| export_method        | app-store                                                               |
| include_bitcode      | true                                                                    |
| project              | Main/Sources/Application/Projects/iOS/********.xcodeproj                |
| scheme               | ********                                                                |
| configuration        | Default                                                                 |
| xcconfig             | Main/Sources/iOS/Application/Config/ausnz/ausnz_appstore.xcconfig       |
| output_directory     | Build/                                                                  |
| output_name          | ********                                                                |
| destination          | generic/platform=iOS                                                    |
| clean                | false                                                                   |
| silent               | false                                                                   |
| use_legacy_build_api | false                                                                   |
| buildlog_path        | ~/Library/Logs/gym                                                      |
| xcode_path           | /Applications/Xcode8.2.1.app                                            |
+----------------------+-------------------------------------------------------------------------+

+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------+
|                                                                                                         Generated Build Command                                                                                                          |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------+
| Option                                                                                                                                                         | Value                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------+
| set -o pipefail &&                                                                                                                                             |                                                                         |
| xcodebuild                                                                                                                                                     |                                                                         |
| -scheme ********                                                                                                                                        |                                                                         |
| -project Main/Sources/Application/Projects/iOS/********.xcodeproj                                                                                             |                                                                         |
| -configuration Default                                                                                                                                         |                                                                         |
| -destination                                                                                                                                                   | generic/platform=iOS                                                    |
| -xcconfig                                                                                                                                                      | Main/Sources/iOS/Application/Config/ausnz/ausnz_appstore.xcconfig |
| -archivePath /Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********.xcarchive |                                                                         |
| archive                                                                                                                                                        |                                                                         |
| | tee /Users/sluptak/Library/Logs/gym/****************.log                                                                                      |                                                                         |
| | xcpretty                                                                                                                                                     |                                                                         |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------+
INFO [2017-04-03 12:55:03.35]: $ set -o pipefail && xcodebuild -scheme ******** -project Main/Sources/Application/Projects/iOS/********.xcodeproj -configuration Default -destination 'generic/platform=iOS' -xcconfig 'Main/Sources/iOS/Application/Config/ausnz/ausnz_appstore.xcconfig' -archivePath /Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********.xcarchive archive | tee /Users/sluptak/Library/Logs/gym/********.log | xcpretty


...
...
...


WARN [2017-04-03 13:10:04.90]: Generated plist file with the following values:
INFO [2017-04-03 13:10:04.90]: ▸ -----------------------------------------
INFO [2017-04-03 13:10:04.90]: ▸ {
INFO [2017-04-03 13:10:04.90]: ▸   "method": "app-store",
INFO [2017-04-03 13:10:04.90]: ▸   "uploadSymbols": true,
INFO [2017-04-03 13:10:04.90]: ▸   "uploadBitcode": true
INFO [2017-04-03 13:10:04.90]: ▸ }
INFO [2017-04-03 13:10:04.90]: ▸ -----------------------------------------
INFO [2017-04-03 13:10:04.90]: This results in the following plist file:
INFO [2017-04-03 13:10:04.90]: ▸ -----------------------------------------
INFO [2017-04-03 13:10:04.90]: ▸ <?xml version="1.0" encoding="UTF-8"?>
INFO [2017-04-03 13:10:04.90]: ▸ <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
INFO [2017-04-03 13:10:04.90]: ▸ <plist version="1.0">
INFO [2017-04-03 13:10:04.90]: ▸ <dict>
INFO [2017-04-03 13:10:04.90]: ▸ 	<key>method</key>
INFO [2017-04-03 13:10:04.90]: ▸ 	<string>app-store</string>
INFO [2017-04-03 13:10:04.90]: ▸ 	<key>uploadBitcode</key>
INFO [2017-04-03 13:10:04.90]: ▸ 	<true/>
INFO [2017-04-03 13:10:04.90]: ▸ 	<key>uploadSymbols</key>
INFO [2017-04-03 13:10:04.90]: ▸ 	<true/>
INFO [2017-04-03 13:10:04.90]: ▸ </dict>
INFO [2017-04-03 13:10:04.90]: ▸ </plist>
INFO [2017-04-03 13:10:04.90]: ▸ -----------------------------------------
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------+
|                                                                                                                Generated Package Command                                                                                                                 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------+
| Option                                                                                                                                                         | Value                                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------+
| /usr/bin/xcrun /Users/sluptak/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.24.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive     |                                                                                         |
| -exportOptionsPlist                                                                                                                                            | /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_config20170403-62905-18pi385.plist |
| -archivePath /Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********.xcarchive |                                                                                         |
| -exportPath                                                                                                                                                    | /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_output20170403-62905-1etz8a2       |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------+
INFO [2017-04-03 13:10:04.91]: $ /usr/bin/xcrun /Users/sluptak/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fastlane-2.24.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_config20170403-62905-18pi385.plist' -archivePath /Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********_ausnz_app-store_2017-04-03-1254_17.0.0.37528\ 2017-04-03\ 12.55.03.xcarchive -exportPath '/var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_output20170403-62905-1etz8a2' 
INFO [2017-04-03 13:10:04.94]: ▸ rbenv detected, removing env variables
INFO [2017-04-03 13:10:04.95]: ▸ rbenv: no such command `shell'
INFO [2017-04-03 13:10:04.95]: ▸ + xcodebuild -exportArchive -exportOptionsPlist /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_config20170403-62905-18pi385.plist -archivePath '/Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********_ausnz_app-store_2017-04-03-1254_17.0.0.37528 2017-04-03 12.55.03.xcarchive' -exportPath /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_output20170403-62905-1etz8a2
INFO [2017-04-03 13:10:05.59]: ▸ 2017-04-03 13:10:05.592 xcodebuild[75674:1458501] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/********_2017-04-03_13-10-05.590.xcdistributionlogs'.
INFO [2017-04-03 13:10:07.03]: ▸ 1.2.840.113635.100.1.61
INFO [2017-04-03 13:10:28.58]: ▸ Exported ********_ausnz_app-store_2017-04-03-1254_17.0.0.37528 2017-04-03 12.55.03.xcarchive to: /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_output20170403-62905-1etz8a2
INFO [2017-04-03 13:10:28.59]: ▸ ** EXPORT SUCCEEDED **
INFO [2017-04-03 13:10:28.69]: Compressing 5 dSYM(s)
INFO [2017-04-03 13:10:28.69]: $ cd '/Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********_ausnz_app-store_2017-04-03-1254_17.0.0.37528 2017-04-03 12.55.03.xcarchive/dSYMs' && zip -r '/Users/sluptak/Dev/2/Build/********_ausnz_app-store_2017-04-03-1254_17.0.0.37528.app.dSYM.zip' *.dSYM
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/Contents/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/Contents/Info.plist (deflated 52%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/Contents/Resources/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/Contents/Resources/DWARF/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********.app.dSYM/Contents/Resources/DWARF/******** (deflated 74%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/Contents/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/Contents/Info.plist (deflated 51%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/Contents/Resources/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/Contents/Resources/DWARF/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Photo.appex.dSYM/Contents/Resources/DWARF/********Photo (deflated 62%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/Contents/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/Contents/Info.plist (deflated 51%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/Contents/Resources/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/Contents/Resources/DWARF/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: ********Widget.appex.dSYM/Contents/Resources/DWARF/********Widget (deflated 65%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/Contents/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/Contents/Info.plist (deflated 51%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/Contents/Resources/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/Contents/Resources/DWARF/ (stored 0%)
INFO [2017-04-03 13:10:31.27]: ▸   adding: WatchKitApp Extension.appex.dSYM/Contents/Resources/DWARF/WatchKitApp Extension (deflated 61%)

INFO [2017-04-03 13:10:31.27]: Successfully exported and compressed dSYM file
INFO [2017-04-03 13:10:31.28]: Successfully exported and signed the ipa file:
INFO [2017-04-03 13:10:31.28]: ********.ipa
INFO [2017-04-03 13:10:31.28]: Cruising back to lane 'appstore' 🚘
INFO [2017-04-03 13:10:31.28]: Successfully generated documentation at path '/Users/sluptak/Dev/****/fastlane/README.md'

+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | Switch to build lane                                          | 0           |
| 5    | gym                                                           | 932         |
+------+---------------------------------------------------------------+-------------+

INFO [2017-04-03 13:10:31.32]: fastlane.tools just saved you 16 minutes! 🎉


```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 10.12.4                                          |
| Ruby                        | 2.3.3                                            |
| Bundler?                    | false                                            |
| Git                         | git version 2.6.3                                |
| Installation Source         | ~/.rbenv/versions/2.3.3/bin/fastlane             |
| Host                        | Mac OS X 10.12.4 (16E195)                        |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.3/lib                      |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode8.2.1.app/Contents/Developer/ |
| Xcode Version               | 8.2.1                                            |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
desc "Builds appstore version for AppStore submission"
desc "Provide 'region' parameter"
lane :appstore do |options|
  options.merge!(flavour: 'app-store')

  build(options)
end

private_lane :build do |options|
  gym(export_method: options[:flavour],
      include_bitcode: true,
      project: "Main/Sources/Application/Projects/iOS/********.xcodeproj",
      scheme: '********',
      configuration: 'Default',
      xcconfig: xcconfig_file_path,
      output_directory: "Build/",
      output_name: '*********')
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.24.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.54.0       |
| faraday                   | 0.11.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.24       |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.2        |
| bundler                   | 1.14.3       |
| faraday_middleware        | 0.11.0.1     |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| io-console                | 0.4.5        |
</details>

*generated on:* **2017-04-03**
</details>

**Comment by EskiMag** [comment]
I also tried `otool -l <PATH_TO_MY_BINARY_IN_THE_IPA> | grep bitcode` and `otool -l <PATH_TO_MY_BINARY_IN_THE_IPA> | grep __LLVM` with zero results.

**Comment by ohayon** [comment]
Hey @EskiMag, if you archive and export your IPA from Xcode itself, does Apple still ping you with the issue? Thanks! 🐙

**Comment by EskiMag** [comment]
Sorry, I forgot to mention this. Of course I tried to archive & export it from Xcode. This way it works properly. Do you have some idea?

**Comment by ohayon** [comment]
Not sure right off the bat, but I'm curious, does running these two commands and then uploading the result work for you?
```
xcodebuild -scheme ******** -project Main/Sources/Application/Projects/iOS/********.xcodeproj -configuration Default -destination 'generic/platform=iOS' -xcconfig 'Main/Sources/iOS/Application/Config/ausnz/ausnz_appstore.xcconfig' -archivePath /Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********.xcarchive archive | tee /Users/sluptak/Library/Logs/gym/********.log
```

```
xcodebuild -exportArchive -exportOptionsPlist /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_config20170403-62905-18pi385.plist -archivePath '/Users/sluptak/Library/Developer/Xcode/Archives/2017-04-03/********_ausnz_app-store_2017-04-03-1254_17.0.0.37528 2017-04-03 12.55.03.xcarchive' -exportPath /var/folders/r4/dknncvn52f75ck_kzq5801180000gn/T/gym_output20170403-62905-1etz8a2```

**Comment by EskiMag** [comment]
@ohayon OK, so I tried to do it with those plain `xcodebuild` and I received email from iTunes Connect, that my WatchKitExtension does not contain bitcode. Interesting. Looks like the problem is somewhere inside the Xcode.

**Comment by EskiMag** [comment]
When I tried to append `OTHER_LDFLAGS=-fembed-bitcode OTHER_CFLAGS=-fembed-bitcode` after the first `xcodebuild` command, I get `can't write output file: /Users/sluptak/Library/Developer/Xcode/DerivedData/**********-ardwmlffacnvhcdhktnabatatkhk/Build/Intermediates/ArchiveIntermediates/**********/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/lib*******.a.strip (Invalid argument)` error and some `/Applications/Xcode8.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning for architecture: arm64 same member name (DetailParserPoiCategoryGroup-A44459DED4065366.o) in output file used for input files: /Users/sluptak/Library/Developer/Xcode/DerivedData/********-ardwmlffacnvhcdhktnabatatkhk/Build/Intermediates/ArchiveIntermediates/********/IntermediateBuildFilesPath/********.build/Default-iphoneos/********.build/Objects-normal/arm64/lib********.a(DetailParserPoiCategoryGroup-A44459DED4065366.o) and: /Users/sluptak/Library/Developer/Xcode/DerivedData/********-ardwmlffacnvhcdhktnabatatkhk/Build/Intermediates/ArchiveIntermediates/********/IntermediateBuildFilesPath/********.build/Default-iphoneos/SygicCore.build/Objects-normal/arm64/lib********.a(DetailParserPoiCategoryGroup-A44459DED4065366.o) due to use of basename, truncation and blank padding` warnings before that.

**Comment by ohayon** [comment]
Hmmmm 🤔. I am not too familiar with this type of issue. I just did a quick search on Stack Overflow and it looks like there are a handful of answers on there like making sure your AppID is different in the dev portal and disabling bitcode all together. 

I am going to close this issue for now as it seems to be an issue with project configuration and Xcode/Apple, but please feel free to comment again on here to continue discussion and we would love to hear what fixes the situation. 🚀 🐙

**Comment by EskiMag** [comment]
OK, so I tried to generate the `xcodearchive` with Xcode (8.2.1) and then I used `gym` to export an IPA from that `xcarchive` and it worked correctly with no warning from iTunes Connect. So the difference is probably between the `xcodebuild` and the Xcode. Do you have an idea how to compare what is actually happening during both methods? I know that `gym` is saving a log for me, and so does the Xcode, but the format of those files is different AFAIK. Do you have some suggestion please?

**Comment by ohayon** [comment]
Yeah! @EskiMag if you do `xcode-select -p` that will show you which version of `xcodebuild` is being used. You could then run `sudo xcode-select -s /path/to/xcode/version/you/want` and then you should be able to match up the `xcodebuild` command to be running the same version as Xcode itself.

Let us know if that is helpful! 🚀

**Comment by EskiMag** [comment]
Yes, I know that command and my `xcode-select` is already pointed to **8.2.1**, also when I am using the Xcode itself, I am using the **8.2.1**. I also tried to uninstall every other Xcode, to prevent some mismatch. So the thing you suggest is probably not the reason for this weird behaviour.

**Comment by ohayon** [comment]
Hmm, I'm not too sure what a good solution would be for that. I would suggest taking a look at Stack Overflow to see if others are seeing the same issue between Xcode itself and `xcodebuild` checked out to the same version.

**Comment by SygicOfficial** [comment]
OK, I have another update. I created dummy empty application with WatchKitApp and WatchKitApp Extension, I replicated Bundle IDs, Versions, Capabilities and then tried `gym` and `deliver` and everything worked perfectly. So there have to be something special in our `xcodeproj` which is quite huge, so I am quite curious what I am going to do now. :-| :-D Wish me good luck...

**Comment by EskiMag** [comment]
OK, I have another update. I created dummy empty application with WatchKitApp and WatchKitApp Extension, I replicated Bundle IDs, Versions, Capabilities and then tried `gym` and `deliver` and everything worked perfectly. So there have to be something special in our xcodeproj which is quite huge, so I am quite curious what I am going to do now. :-| :-D Wish me good luck...

**Comment by ohayon** [comment]
Best of luck @EskiMag ! Thanks for the follow up! We would love to hear what ends up fixing it!

**Comment by EskiMag** [comment]
OK. It seems I have found the solution. I had `ENABLE_BITCODE = NO` setting in my `.xcconfig` file and then explicitly set `ENABLE_BITCODE = YES` for WatchKitApp and WatchKitExtension. Somehow, it was resolved differently when using Xcode and when using `xcodebuild` ... it's really weird. Maybe I'll try later to investigate this issue a bit deeper, but for now it's working for me with fastlane and that's important at the moment. Thanks everybody for your support and advices. :-D
